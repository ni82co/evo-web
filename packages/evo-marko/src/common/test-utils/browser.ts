import { fireEvent, createEvent } from "@marko/testing-library";
/**
 * Simulates a touch based scroll event over 4 animation frames.
 *
 * @param {HTMLElement} el The element to scroll.
 * @param {number} to The new scrollLeft for the element.
 * @param {function} cb A callback to call after the scroll.
 */
function simulateScroll(el: HTMLElement, to: number, cb: Function) {
  fireEvent.scroll(el);
  el.scrollLeft = to;
  setTimeout(cb, 600);
}

function waitFrames(count: number, cb: Function) {
  if (count) {
    return requestAnimationFrame(() => {
      waitFrames(count - 1, cb);
    });
  }

  cb();
}

interface FastAnimations {
  fastAnimationStyle?: HTMLStyleElement;
  start: () => void;
  stop: () => void;
}

const fastAnimations:FastAnimations = {
  // Adds an style to the document which forces all transitions to run more quickly for the tests.
  start() {
    if (this.fastAnimationStyle) {
      return;
    }

    this.fastAnimationStyle = document.createElement("style");
    this.fastAnimationStyle.innerHTML = `* {
                transition-duration: 0.1s !important;
                transition-delay: 0.1s !important;
                scroll-behavior: auto !important;
            }`;
    document.head.appendChild(this.fastAnimationStyle);
  },
  stop() {
    if (!this.fastAnimationStyle) {
      return;
    }

    document.head.removeChild(this.fastAnimationStyle);
    this.fastAnimationStyle = undefined;
  },
};

export { fastAnimations, waitFrames, simulateScroll };
