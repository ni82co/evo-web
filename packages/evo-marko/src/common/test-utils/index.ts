export * from "./browser"
export * from "./snapshots";