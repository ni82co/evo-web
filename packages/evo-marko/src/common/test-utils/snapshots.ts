import { render, normalize } from "@marko/testing-library";
import { expect } from "vitest";

async function snapshotHTML(template: any, input?: any) {
  const { container } = await render(template, input);
  expect(normalize(container)).toMatchSnapshot();
}

export { snapshotHTML };
