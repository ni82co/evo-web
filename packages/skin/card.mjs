import './dist/card/card.css';
