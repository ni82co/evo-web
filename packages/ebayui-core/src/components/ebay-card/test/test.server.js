import { it } from "vitest";
import { composeStories } from "@storybook/marko";
import { snapshotHTML } from "../../../common/test-utils/snapshots";
import * as stories from "../card.stories"; // import all stories from the stories file

const { Default, Anchor, Button, Minimum } = composeStories(stories);
const htmlSnap = snapshotHTML(__dirname);

describe("ebay-card", () => {
    it("renders default", async () => {
        await htmlSnap(Default);
    });

    it("renders with layout=vertical", async () => {
        await htmlSnap(Default, { layout: "vertical" });
    });

    it("renders anchor", async () => {
        await htmlSnap(Anchor);
    });

    it("renders button", async () => {
        await htmlSnap(Button);
    });

    it("renders minimum", async () => {
        await htmlSnap(Minimum);
    });

    it("renders anchor disabled", async () => {
        await htmlSnap(Anchor, { disabled: true });
    });

    it("renders button disabled", async () => {
        await htmlSnap(Button, { disabled: true });
    });
});
