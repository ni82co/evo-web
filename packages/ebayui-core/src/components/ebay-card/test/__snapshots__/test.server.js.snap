// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ebay-card > renders anchor 1`] = `
"[36m<DocumentFragment>[39m
  [36m<a[39m
    [33mclass[39m=[32m"card"[39m
    [33mhref[39m=[32m"https://www.ebay.com"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"card__media"[39m
    [36m>[39m
      [36m<img[39m
        [33malt[39m=[32m"Card"[39m
        [33msrc[39m=[32m"https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"[39m
      [36m/>[39m
    [36m</div>[39m
    [36m<div[39m
      [33mclass[39m=[32m"card__main"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m"card__overline"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--16"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-authenticity-guarantee-16"[39m
              [33mviewBox[39m=[32m"0 0 16 16"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M11.207 5.793a1 1 0 0 1 0 1.414l-3.5 3.5a1 1 0 0 1-1.414 0l-1.5-1.5a1 1 0 0 1 1.414-1.414L7 8.586l2.793-2.793a1 1 0 0 1 1.414 0Z"[39m
              [36m/>[39m
              [36m<path[39m
                [33mclip-rule[39m=[32m"evenodd"[39m
                [33md[39m=[32m"M13.657 2.343a3.32 3.32 0 0 1 .976 2.3c.007.387.163.764.432 1.042C15.644 6.285 16 7.1 16 8c0 .9-.356 1.716-.936 2.316a1.547 1.547 0 0 0-.431 1.042 3.322 3.322 0 0 1-.976 2.299 3.322 3.322 0 0 1-2.3.976 1.548 1.548 0 0 0-1.041.431A3.32 3.32 0 0 1 8 16a3.32 3.32 0 0 1-2.316-.936 1.547 1.547 0 0 0-1.041-.431 3.322 3.322 0 0 1-2.3-.976 3.322 3.322 0 0 1-.975-2.299 1.548 1.548 0 0 0-.432-1.042A3.322 3.322 0 0 1 0 8c0-.9.357-1.716.936-2.316.269-.278.425-.655.432-1.042.014-.833.34-1.663.975-2.299a3.322 3.322 0 0 1 2.3-.976 1.547 1.547 0 0 0 1.04-.431A3.324 3.324 0 0 1 8 0c.9 0 1.716.357 2.316.936.278.269.655.425 1.042.432.834.014 1.663.34 2.3.975Zm-9.9 1.414a1.322 1.322 0 0 0-.39.92 3.547 3.547 0 0 1-.993 2.397A1.322 1.322 0 0 0 2 8c0 .361.14.685.374.927.621.642.978 1.503.994 2.396.005.337.135.665.39.92s.583.384.92.39a3.547 3.547 0 0 1 2.396.993c.242.234.565.374.926.374.36 0 .684-.14.926-.374a3.548 3.548 0 0 1 2.397-.993c.336-.006.665-.135.92-.39s.384-.584.39-.92a3.547 3.547 0 0 1 .993-2.397C13.86 8.685 14 8.361 14 8c0-.36-.14-.684-.374-.926a3.548 3.548 0 0 1-.993-2.397 1.34 1.34 0 0 0-.39-.92 1.322 1.322 0 0 0-.92-.39 3.548 3.548 0 0 1-2.397-.993A1.322 1.322 0 0 0 8 2c-.36 0-.685.14-.926.374a3.547 3.547 0 0 1-2.397.993 1.322 1.322 0 0 0-.92.39Z"[39m
                [33mfill-rule[39m=[32m"evenodd"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-authenticity-guarantee-16"[39m
          [36m/>[39m
        [36m</svg>[39m
        [0m Authenticity Guaranteed[0m
      [36m</div>[39m
      [36m<h3[39m
        [33mclass[39m=[32m"card__title"[39m
      [36m>[39m
        [0mAuthentic Rookie Cards[0m
      [36m</h3>[39m
      [36m<div[39m
        [33mclass[39m=[32m"card__description"[39m
      [36m>[39m
        [0mProfessional-grade rookie cards that appreciate faster than most people's cooking skills and last longer than New Year's resolutions.[0m
      [36m</div>[39m
    [36m</div>[39m
  [36m</a>[39m
[36m</DocumentFragment>[39m"
`;

exports[`ebay-card > renders anchor disabled 1`] = `
"[36m<DocumentFragment>[39m
  [36m<a[39m
    [33mclass[39m=[32m"card"[39m
    [33mhref[39m=[32m""[39m
    [33mtabindex[39m=[32m"-1"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"card__media"[39m
    [36m>[39m
      [36m<img[39m
        [33malt[39m=[32m"Card"[39m
        [33msrc[39m=[32m"https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"[39m
      [36m/>[39m
    [36m</div>[39m
    [36m<div[39m
      [33mclass[39m=[32m"card__main"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m"card__overline"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--16"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-authenticity-guarantee-16"[39m
              [33mviewBox[39m=[32m"0 0 16 16"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M11.207 5.793a1 1 0 0 1 0 1.414l-3.5 3.5a1 1 0 0 1-1.414 0l-1.5-1.5a1 1 0 0 1 1.414-1.414L7 8.586l2.793-2.793a1 1 0 0 1 1.414 0Z"[39m
              [36m/>[39m
              [36m<path[39m
                [33mclip-rule[39m=[32m"evenodd"[39m
                [33md[39m=[32m"M13.657 2.343a3.32 3.32 0 0 1 .976 2.3c.007.387.163.764.432 1.042C15.644 6.285 16 7.1 16 8c0 .9-.356 1.716-.936 2.316a1.547 1.547 0 0 0-.431 1.042 3.322 3.322 0 0 1-.976 2.299 3.322 3.322 0 0 1-2.3.976 1.548 1.548 0 0 0-1.041.431A3.32 3.32 0 0 1 8 16a3.32 3.32 0 0 1-2.316-.936 1.547 1.547 0 0 0-1.041-.431 3.322 3.322 0 0 1-2.3-.976 3.322 3.322 0 0 1-.975-2.299 1.548 1.548 0 0 0-.432-1.042A3.322 3.322 0 0 1 0 8c0-.9.357-1.716.936-2.316.269-.278.425-.655.432-1.042.014-.833.34-1.663.975-2.299a3.322 3.322 0 0 1 2.3-.976 1.547 1.547 0 0 0 1.04-.431A3.324 3.324 0 0 1 8 0c.9 0 1.716.357 2.316.936.278.269.655.425 1.042.432.834.014 1.663.34 2.3.975Zm-9.9 1.414a1.322 1.322 0 0 0-.39.92 3.547 3.547 0 0 1-.993 2.397A1.322 1.322 0 0 0 2 8c0 .361.14.685.374.927.621.642.978 1.503.994 2.396.005.337.135.665.39.92s.583.384.92.39a3.547 3.547 0 0 1 2.396.993c.242.234.565.374.926.374.36 0 .684-.14.926-.374a3.548 3.548 0 0 1 2.397-.993c.336-.006.665-.135.92-.39s.384-.584.39-.92a3.547 3.547 0 0 1 .993-2.397C13.86 8.685 14 8.361 14 8c0-.36-.14-.684-.374-.926a3.548 3.548 0 0 1-.993-2.397 1.34 1.34 0 0 0-.39-.92 1.322 1.322 0 0 0-.92-.39 3.548 3.548 0 0 1-2.397-.993A1.322 1.322 0 0 0 8 2c-.36 0-.685.14-.926.374a3.547 3.547 0 0 1-2.397.993 1.322 1.322 0 0 0-.92.39Z"[39m
                [33mfill-rule[39m=[32m"evenodd"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-authenticity-guarantee-16"[39m
          [36m/>[39m
        [36m</svg>[39m
        [0m Authenticity Guaranteed[0m
      [36m</div>[39m
      [36m<h3[39m
        [33mclass[39m=[32m"card__title"[39m
      [36m>[39m
        [0mAuthentic Rookie Cards[0m
      [36m</h3>[39m
      [36m<div[39m
        [33mclass[39m=[32m"card__description"[39m
      [36m>[39m
        [0mProfessional-grade rookie cards that appreciate faster than most people's cooking skills and last longer than New Year's resolutions.[0m
      [36m</div>[39m
    [36m</div>[39m
  [36m</a>[39m
[36m</DocumentFragment>[39m"
`;

exports[`ebay-card > renders button 1`] = `
"[36m<DocumentFragment>[39m
  [36m<button[39m
    [33mclass[39m=[32m"card"[39m
    [33mtype[39m=[32m"button"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"card__media"[39m
    [36m>[39m
      [36m<img[39m
        [33malt[39m=[32m"Card"[39m
        [33msrc[39m=[32m"https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"[39m
      [36m/>[39m
    [36m</div>[39m
    [36m<div[39m
      [33mclass[39m=[32m"card__main"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m"card__overline"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--16"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-authenticity-guarantee-16"[39m
              [33mviewBox[39m=[32m"0 0 16 16"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M11.207 5.793a1 1 0 0 1 0 1.414l-3.5 3.5a1 1 0 0 1-1.414 0l-1.5-1.5a1 1 0 0 1 1.414-1.414L7 8.586l2.793-2.793a1 1 0 0 1 1.414 0Z"[39m
              [36m/>[39m
              [36m<path[39m
                [33mclip-rule[39m=[32m"evenodd"[39m
                [33md[39m=[32m"M13.657 2.343a3.32 3.32 0 0 1 .976 2.3c.007.387.163.764.432 1.042C15.644 6.285 16 7.1 16 8c0 .9-.356 1.716-.936 2.316a1.547 1.547 0 0 0-.431 1.042 3.322 3.322 0 0 1-.976 2.299 3.322 3.322 0 0 1-2.3.976 1.548 1.548 0 0 0-1.041.431A3.32 3.32 0 0 1 8 16a3.32 3.32 0 0 1-2.316-.936 1.547 1.547 0 0 0-1.041-.431 3.322 3.322 0 0 1-2.3-.976 3.322 3.322 0 0 1-.975-2.299 1.548 1.548 0 0 0-.432-1.042A3.322 3.322 0 0 1 0 8c0-.9.357-1.716.936-2.316.269-.278.425-.655.432-1.042.014-.833.34-1.663.975-2.299a3.322 3.322 0 0 1 2.3-.976 1.547 1.547 0 0 0 1.04-.431A3.324 3.324 0 0 1 8 0c.9 0 1.716.357 2.316.936.278.269.655.425 1.042.432.834.014 1.663.34 2.3.975Zm-9.9 1.414a1.322 1.322 0 0 0-.39.92 3.547 3.547 0 0 1-.993 2.397A1.322 1.322 0 0 0 2 8c0 .361.14.685.374.927.621.642.978 1.503.994 2.396.005.337.135.665.39.92s.583.384.92.39a3.547 3.547 0 0 1 2.396.993c.242.234.565.374.926.374.36 0 .684-.14.926-.374a3.548 3.548 0 0 1 2.397-.993c.336-.006.665-.135.92-.39s.384-.584.39-.92a3.547 3.547 0 0 1 .993-2.397C13.86 8.685 14 8.361 14 8c0-.36-.14-.684-.374-.926a3.548 3.548 0 0 1-.993-2.397 1.34 1.34 0 0 0-.39-.92 1.322 1.322 0 0 0-.92-.39 3.548 3.548 0 0 1-2.397-.993A1.322 1.322 0 0 0 8 2c-.36 0-.685.14-.926.374a3.547 3.547 0 0 1-2.397.993 1.322 1.322 0 0 0-.92.39Z"[39m
                [33mfill-rule[39m=[32m"evenodd"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-authenticity-guarantee-16"[39m
          [36m/>[39m
        [36m</svg>[39m
        [0m Authenticity Guaranteed[0m
      [36m</div>[39m
      [36m<h3[39m
        [33mclass[39m=[32m"card__title"[39m
      [36m>[39m
        [0mAuthentic Rookie Cards[0m
      [36m</h3>[39m
      [36m<div[39m
        [33mclass[39m=[32m"card__description"[39m
      [36m>[39m
        [0mProfessional-grade rookie cards that appreciate faster than most people's cooking skills and last longer than New Year's resolutions.[0m
      [36m</div>[39m
    [36m</div>[39m
  [36m</button>[39m
[36m</DocumentFragment>[39m"
`;

exports[`ebay-card > renders button disabled 1`] = `
"[36m<DocumentFragment>[39m
  [36m<button[39m
    [33mclass[39m=[32m"card"[39m
    [33mdisabled[39m=[32m""[39m
    [33mtabindex[39m=[32m"-1"[39m
    [33mtype[39m=[32m"button"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"card__media"[39m
    [36m>[39m
      [36m<img[39m
        [33malt[39m=[32m"Card"[39m
        [33msrc[39m=[32m"https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"[39m
      [36m/>[39m
    [36m</div>[39m
    [36m<div[39m
      [33mclass[39m=[32m"card__main"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m"card__overline"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--16"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-authenticity-guarantee-16"[39m
              [33mviewBox[39m=[32m"0 0 16 16"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M11.207 5.793a1 1 0 0 1 0 1.414l-3.5 3.5a1 1 0 0 1-1.414 0l-1.5-1.5a1 1 0 0 1 1.414-1.414L7 8.586l2.793-2.793a1 1 0 0 1 1.414 0Z"[39m
              [36m/>[39m
              [36m<path[39m
                [33mclip-rule[39m=[32m"evenodd"[39m
                [33md[39m=[32m"M13.657 2.343a3.32 3.32 0 0 1 .976 2.3c.007.387.163.764.432 1.042C15.644 6.285 16 7.1 16 8c0 .9-.356 1.716-.936 2.316a1.547 1.547 0 0 0-.431 1.042 3.322 3.322 0 0 1-.976 2.299 3.322 3.322 0 0 1-2.3.976 1.548 1.548 0 0 0-1.041.431A3.32 3.32 0 0 1 8 16a3.32 3.32 0 0 1-2.316-.936 1.547 1.547 0 0 0-1.041-.431 3.322 3.322 0 0 1-2.3-.976 3.322 3.322 0 0 1-.975-2.299 1.548 1.548 0 0 0-.432-1.042A3.322 3.322 0 0 1 0 8c0-.9.357-1.716.936-2.316.269-.278.425-.655.432-1.042.014-.833.34-1.663.975-2.299a3.322 3.322 0 0 1 2.3-.976 1.547 1.547 0 0 0 1.04-.431A3.324 3.324 0 0 1 8 0c.9 0 1.716.357 2.316.936.278.269.655.425 1.042.432.834.014 1.663.34 2.3.975Zm-9.9 1.414a1.322 1.322 0 0 0-.39.92 3.547 3.547 0 0 1-.993 2.397A1.322 1.322 0 0 0 2 8c0 .361.14.685.374.927.621.642.978 1.503.994 2.396.005.337.135.665.39.92s.583.384.92.39a3.547 3.547 0 0 1 2.396.993c.242.234.565.374.926.374.36 0 .684-.14.926-.374a3.548 3.548 0 0 1 2.397-.993c.336-.006.665-.135.92-.39s.384-.584.39-.92a3.547 3.547 0 0 1 .993-2.397C13.86 8.685 14 8.361 14 8c0-.36-.14-.684-.374-.926a3.548 3.548 0 0 1-.993-2.397 1.34 1.34 0 0 0-.39-.92 1.322 1.322 0 0 0-.92-.39 3.548 3.548 0 0 1-2.397-.993A1.322 1.322 0 0 0 8 2c-.36 0-.685.14-.926.374a3.547 3.547 0 0 1-2.397.993 1.322 1.322 0 0 0-.92.39Z"[39m
                [33mfill-rule[39m=[32m"evenodd"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-authenticity-guarantee-16"[39m
          [36m/>[39m
        [36m</svg>[39m
        [0m Authenticity Guaranteed[0m
      [36m</div>[39m
      [36m<h3[39m
        [33mclass[39m=[32m"card__title"[39m
      [36m>[39m
        [0mAuthentic Rookie Cards[0m
      [36m</h3>[39m
      [36m<div[39m
        [33mclass[39m=[32m"card__description"[39m
      [36m>[39m
        [0mProfessional-grade rookie cards that appreciate faster than most people's cooking skills and last longer than New Year's resolutions.[0m
      [36m</div>[39m
    [36m</div>[39m
  [36m</button>[39m
[36m</DocumentFragment>[39m"
`;

exports[`ebay-card > renders default 1`] = `
"[36m<DocumentFragment>[39m
  [36m<span[39m
    [33mclass[39m=[32m"card"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"card__media"[39m
    [36m>[39m
      [36m<img[39m
        [33malt[39m=[32m"Card"[39m
        [33msrc[39m=[32m"https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"[39m
      [36m/>[39m
    [36m</div>[39m
    [36m<div[39m
      [33mclass[39m=[32m"card__main"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m"card__overline"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--16"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-authenticity-guarantee-16"[39m
              [33mviewBox[39m=[32m"0 0 16 16"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M11.207 5.793a1 1 0 0 1 0 1.414l-3.5 3.5a1 1 0 0 1-1.414 0l-1.5-1.5a1 1 0 0 1 1.414-1.414L7 8.586l2.793-2.793a1 1 0 0 1 1.414 0Z"[39m
              [36m/>[39m
              [36m<path[39m
                [33mclip-rule[39m=[32m"evenodd"[39m
                [33md[39m=[32m"M13.657 2.343a3.32 3.32 0 0 1 .976 2.3c.007.387.163.764.432 1.042C15.644 6.285 16 7.1 16 8c0 .9-.356 1.716-.936 2.316a1.547 1.547 0 0 0-.431 1.042 3.322 3.322 0 0 1-.976 2.299 3.322 3.322 0 0 1-2.3.976 1.548 1.548 0 0 0-1.041.431A3.32 3.32 0 0 1 8 16a3.32 3.32 0 0 1-2.316-.936 1.547 1.547 0 0 0-1.041-.431 3.322 3.322 0 0 1-2.3-.976 3.322 3.322 0 0 1-.975-2.299 1.548 1.548 0 0 0-.432-1.042A3.322 3.322 0 0 1 0 8c0-.9.357-1.716.936-2.316.269-.278.425-.655.432-1.042.014-.833.34-1.663.975-2.299a3.322 3.322 0 0 1 2.3-.976 1.547 1.547 0 0 0 1.04-.431A3.324 3.324 0 0 1 8 0c.9 0 1.716.357 2.316.936.278.269.655.425 1.042.432.834.014 1.663.34 2.3.975Zm-9.9 1.414a1.322 1.322 0 0 0-.39.92 3.547 3.547 0 0 1-.993 2.397A1.322 1.322 0 0 0 2 8c0 .361.14.685.374.927.621.642.978 1.503.994 2.396.005.337.135.665.39.92s.583.384.92.39a3.547 3.547 0 0 1 2.396.993c.242.234.565.374.926.374.36 0 .684-.14.926-.374a3.548 3.548 0 0 1 2.397-.993c.336-.006.665-.135.92-.39s.384-.584.39-.92a3.547 3.547 0 0 1 .993-2.397C13.86 8.685 14 8.361 14 8c0-.36-.14-.684-.374-.926a3.548 3.548 0 0 1-.993-2.397 1.34 1.34 0 0 0-.39-.92 1.322 1.322 0 0 0-.92-.39 3.548 3.548 0 0 1-2.397-.993A1.322 1.322 0 0 0 8 2c-.36 0-.685.14-.926.374a3.547 3.547 0 0 1-2.397.993 1.322 1.322 0 0 0-.92.39Z"[39m
                [33mfill-rule[39m=[32m"evenodd"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-authenticity-guarantee-16"[39m
          [36m/>[39m
        [36m</svg>[39m
        [0m Authenticity Guaranteed[0m
      [36m</div>[39m
      [36m<h3[39m
        [33mclass[39m=[32m"card__title"[39m
      [36m>[39m
        [0mAuthentic Rookie Cards[0m
      [36m</h3>[39m
      [36m<div[39m
        [33mclass[39m=[32m"card__description"[39m
      [36m>[39m
        [0mProfessional-grade rookie cards that appreciate faster than most people's cooking skills and last longer than New Year's resolutions.[0m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m"card__action"[39m
      [36m>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--tertiary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mSee Details[0m
        [36m</button>[39m
      [36m</div>[39m
    [36m</div>[39m
  [36m</span>[39m
[36m</DocumentFragment>[39m"
`;

exports[`ebay-card > renders minimum 1`] = `
"[36m<DocumentFragment>[39m
  [36m<button[39m
    [33mclass[39m=[32m"card"[39m
    [33mtype[39m=[32m"button"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"card__media"[39m
    [36m>[39m
      [36m<img[39m
        [33malt[39m=[32m"Card"[39m
        [33msrc[39m=[32m"https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"[39m
      [36m/>[39m
    [36m</div>[39m
    [36m<div[39m
      [33mclass[39m=[32m"card__main"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m"card__description"[39m
      [36m>[39m
        [0mProfessional-grade rookie cards that appreciate faster than most people's cooking skills and last longer than New Year's resolutions.[0m
      [36m</div>[39m
    [36m</div>[39m
  [36m</button>[39m
[36m</DocumentFragment>[39m"
`;

exports[`ebay-card > renders with layout=vertical 1`] = `
"[36m<DocumentFragment>[39m
  [36m<span[39m
    [33mclass[39m=[32m"card"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"card__media"[39m
    [36m>[39m
      [36m<img[39m
        [33malt[39m=[32m"Card"[39m
        [33msrc[39m=[32m"https://ir.ebaystatic.com/cr/v/c01/skin/docs/tb-real-square-pic.jpg"[39m
      [36m/>[39m
    [36m</div>[39m
    [36m<div[39m
      [33mclass[39m=[32m"card__main"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m"card__overline"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--16"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-authenticity-guarantee-16"[39m
              [33mviewBox[39m=[32m"0 0 16 16"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M11.207 5.793a1 1 0 0 1 0 1.414l-3.5 3.5a1 1 0 0 1-1.414 0l-1.5-1.5a1 1 0 0 1 1.414-1.414L7 8.586l2.793-2.793a1 1 0 0 1 1.414 0Z"[39m
              [36m/>[39m
              [36m<path[39m
                [33mclip-rule[39m=[32m"evenodd"[39m
                [33md[39m=[32m"M13.657 2.343a3.32 3.32 0 0 1 .976 2.3c.007.387.163.764.432 1.042C15.644 6.285 16 7.1 16 8c0 .9-.356 1.716-.936 2.316a1.547 1.547 0 0 0-.431 1.042 3.322 3.322 0 0 1-.976 2.299 3.322 3.322 0 0 1-2.3.976 1.548 1.548 0 0 0-1.041.431A3.32 3.32 0 0 1 8 16a3.32 3.32 0 0 1-2.316-.936 1.547 1.547 0 0 0-1.041-.431 3.322 3.322 0 0 1-2.3-.976 3.322 3.322 0 0 1-.975-2.299 1.548 1.548 0 0 0-.432-1.042A3.322 3.322 0 0 1 0 8c0-.9.357-1.716.936-2.316.269-.278.425-.655.432-1.042.014-.833.34-1.663.975-2.299a3.322 3.322 0 0 1 2.3-.976 1.547 1.547 0 0 0 1.04-.431A3.324 3.324 0 0 1 8 0c.9 0 1.716.357 2.316.936.278.269.655.425 1.042.432.834.014 1.663.34 2.3.975Zm-9.9 1.414a1.322 1.322 0 0 0-.39.92 3.547 3.547 0 0 1-.993 2.397A1.322 1.322 0 0 0 2 8c0 .361.14.685.374.927.621.642.978 1.503.994 2.396.005.337.135.665.39.92s.583.384.92.39a3.547 3.547 0 0 1 2.396.993c.242.234.565.374.926.374.36 0 .684-.14.926-.374a3.548 3.548 0 0 1 2.397-.993c.336-.006.665-.135.92-.39s.384-.584.39-.92a3.547 3.547 0 0 1 .993-2.397C13.86 8.685 14 8.361 14 8c0-.36-.14-.684-.374-.926a3.548 3.548 0 0 1-.993-2.397 1.34 1.34 0 0 0-.39-.92 1.322 1.322 0 0 0-.92-.39 3.548 3.548 0 0 1-2.397-.993A1.322 1.322 0 0 0 8 2c-.36 0-.685.14-.926.374a3.547 3.547 0 0 1-2.397.993 1.322 1.322 0 0 0-.92.39Z"[39m
                [33mfill-rule[39m=[32m"evenodd"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-authenticity-guarantee-16"[39m
          [36m/>[39m
        [36m</svg>[39m
        [0m Authenticity Guaranteed[0m
      [36m</div>[39m
      [36m<h3[39m
        [33mclass[39m=[32m"card__title"[39m
      [36m>[39m
        [0mAuthentic Rookie Cards[0m
      [36m</h3>[39m
      [36m<div[39m
        [33mclass[39m=[32m"card__description"[39m
      [36m>[39m
        [0mProfessional-grade rookie cards that appreciate faster than most people's cooking skills and last longer than New Year's resolutions.[0m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m"card__action"[39m
      [36m>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn btn--fluid btn--tertiary"[39m
          [33mdata-ebayui[39m=[32m""[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [0mSee Details[0m
        [36m</button>[39m
      [36m</div>[39m
    [36m</div>[39m
  [36m</span>[39m
[36m</DocumentFragment>[39m"
`;
